src/library/Text/Text.js

```diff
@@ -0,0 +1,165 @@
+    appearance: propAppearance,
+    color,
+    element,
+    fontWeight,
+    noMargins,
+    theme: baseTheme
+  }) => {
+    let theme = componentTheme(baseTheme);
+    const headingValues = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
+    const isHeading = headingValues.indexOf(element) !== -1;
+    const appearance = propAppearance
+      ? propAppearance
+      : isHeading ? element : null;
+    const isHeadingAppearance = headingValues.indexOf(appearance) !== -1;
+    const rtl = theme.direction === 'rtl';
+
+    if (isHeadingAppearance && appearance) {
+      theme = {
+        ...theme,
+        Text_color: theme[`Text_color_${appearance}`],
+        Text_fontSize: theme[`Text_fontSize_${appearance}`],
+        Text_fontWeight: theme[`Text_fontWeight_${appearance}`],
+        Text_lineHeight:
+          ['h5', 'h6'].indexOf(appearance) !== -1
+            ? theme.Text_lineHeight_headingSmall
+            : theme.Text_lineHeight_heading,
+        Text_marginBottom: theme.Text_marginBottom_heading
+      };
+    }
+
+    let styles = {
+      color: color || theme.Text_color,
+      fontSize:
+        appearance && !isHeadingAppearance
+          ? theme[`Text_fontSize_${appearance}`]
+          : theme.Text_fontSize,
+      fontWeight: (() => {
+        if (fontWeight && theme[`fontWeight_${fontWeight}`]) {
+          return theme[`fontWeight_${fontWeight}`];
+        } else if (isHeadingAppearance && theme.Text_fontWeight) {
+          return theme.Text_fontWeight;
+        } else {
+          return fontWeight || theme.fontWeight_regular;
+        }
+      })(),
+      lineHeight: theme.Text_lineHeight,
+      // Not normalized because we actually want `##em` as applied value
+      marginBottom: !noMargins ? theme.Text_marginBottom : 0,
+      marginTop: 0,
+      textAlign: (() => {
+        if ((rtl && align == 'start') || (!rtl && align == 'end')) {
+          return 'right';
+        } else if ((rtl && align == 'end') || align == 'start') {
+          return 'left';
+        } else {
+          return align;
+        }
+      })(),
+
+      '&::selection': {
+        backgroundColor: theme.Text_backgroundColor_selected
+      }
+    };
+
+    if (['code', 'kbd', 'samp'].indexOf(element) !== -1) {
+      (styles: Object).fontFamily = theme.fontFamily_monospace;
+    }
+
+    return styles;
+  }
+};
+
+// Text's root node must be created outside of render, so that the entire DOM
+// element is replaced only when the element prop is changed, otherwise it is
+// updated in place
+function createRootNode(props: Props) {
+  const { element = Text.defaultProps.element } = props;
+
+  return createStyledComponent(element, styles.root, {
+    includeStyleReset: true,
+    rootEl: element
+  });
+}
+
+/**
+ * The Text component provides styles and semantic meaning for text and headings
+ * in a manner consistent with other components.
+ */
+export default class Text extends Component<Props> {
+  static defaultProps = {
+    align: 'start',
```

Comment:
I am not sure that we should have a default value for the align prop.  To me, it seems that not applying a default alignment would be better.  Blocks of text would then be more easily controlled by a parent layout component.

Reply:
Great suggestion.